src/main/java/com/imsweb/mph/mpgroups/Mp2010HematopoieticGroup.java

```diff
@@ -0,0 +1,473 @@
+                MphRuleResult result = new MphRuleResult();
+                String hist1 = i1.getHistology(), hist2 = i2.getHistology(), site1 = i1.getPrimarySite(), site2 = i2.getPrimarySite();
+                boolean sameLocation = site1.equals(site2) || (site1.substring(0, 3).equals(site2.substring(0, 3)) && !MphConstants.LYMPH_NODE.equals(site1.substring(0, 3)));
+                if (hist1.equals(hist2) || !MphConstants.LYMPHOMA_NOS_AND_NON_HODGKIN_LYMPHOMA.containsAll(Arrays.asList(hist1, hist2)) || !sameLocation)
+                    result.setResult(MphUtils.RuleResult.FALSE);
+                else {
+                    int simultaneouslyPresent = GroupUtility.verifyDaysApart(i1, i2, 21);
+                    if (simultaneouslyPresent == -1) {
+                        result.setResult(MphUtils.RuleResult.UNKNOWN);
+                        result.setMessage("Unable to apply Rule" + this.getStep() + " of " + this.getGroupId() + ". Known diagnosis date should be provided.");
+                    }
+                    else
+                        result.setResult(0 == simultaneouslyPresent ? MphUtils.RuleResult.TRUE : MphUtils.RuleResult.FALSE);
+                }
+
+                return result;
+            }
+        };
+        rule.setReason("Abstract a single primary when two or more types of non-Hodgkin lymphoma are simultaneously present in the same anatomic location(s), such "
+                + "as the same lymph node or lymph node region(s), the same organ(s), and/or the same tissue(s)");
+        rule.getNotes().add("For the purpose of using the rules, a non-Hodgkin lymphoma is any lymphoma (including the leukemia/lymphomas) not stated to be Hodgkin "
+                + "lymphoma, NOS or a type of Hodgkin lymphoma.\n • Hodgkin lymphomas are: 9650/-9653/3, 9655/3, 9659/3, 9663/3");
+        rule.getNotes().add("Use Rule M15 for simultaneous occurrences of two or more cutaneous lymphomas. Do not use this rule for cutaneous lymphomas. Simultaneous "
+                + "occurrences of two or more cutaneous lymphomas, other than an NOS and more specific, are extremely rare. If there are simultaneous cutaneous "
+                + "lymphomas, DO NOT use this rule; proceed to rule M15 (use Multiple Primaries Calculator)");
+        rule.getNotes().add("When the neoplasm is in an early stage, the involved lymph node(s) will be in the same region as defined by ICD-O-3 codes. See Appendix C for help "
+                + "identifying lymph node names, chains, regions and codes.");
+        rule.getNotes().add("When the neoplasm is in a more advanced stage, both non-Hodgkin lymphomas may be present in multiple lymph nodes in the same regions as defined "
+                + "by ICD-O-3, or in an organ and that organ’s regional lymph nodes, or in multiple organs.\n"
+                + "• Although the combination of two or more types of non-Hodgkin lymphoma must be present in each of the involved sites in order to abstract as a "
```

Comment:
There is a bullet in the text that looks like a non-ASCII character. Can you please check all your rules, make sure you only use ASCII. If they use bullets in their descriptions, please replace them with dashes. If your descriptions are in HTML and not just text, then you can use the list tags (but it doesn't look like it's HTML).